File: 383.赎金信.cpp
/*
 * @lc app=leetcode.cn id=383 lang=cpp
 *
 * [383] 赎金信
 */

// @lc code=start
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        if(ransomNote.length() == 0) return true;
        if(magazine.length() == 0) return false;
        int i = 0, j = 0, k = 0;
        while(i < ransomNote.length())
        {
            j = 0;
            k = 0;
            while(j < magazine.length())
            {
                if(ransomNote[i] == magazine[j])
                {
                    magazine[j] = '\0';
                    k = 1;
                    break;
                }
                j++;
            }
            if(k == 0) return false;
            i++;
        }
        return true;
    }
};
// @lc code=end

File: Answer/383.赎金信.cs
/*
 * @lc app=leetcode.cn id=383 lang=csharp
 *
 * [383] 赎金信
 */

// @lc code=start
public class Solution {
    public bool CanConstruct(string ransomNote, string magazine) {
        if(ransomNote.Length == 0) return true;
        if(magazine.Length == 0) return false;
        for(int i = 0; i < ransomNote.Length; i++)
        {
            if(ransomNote[i] == '|') continue;
            int n1 = 0, n2 = 0;
            for(int j = i; j < ransomNote.Length; j++)
            {
                if(ransomNote[j] == ransomNote[i])
                {
                    n1++;
                }
            }
            for(int k = 0; k < magazine.Length; k++)
            {
                if(magazine[k] == ransomNote[i])
                {
                    n2++;
                }
            }
            if(n1 > n2) return false;
            ransomNote = ransomNote.Replace(ransomNote[i], '|');
            magazine = magazine.Replace(ransomNote[i], '|');
        }
        return true;
    }
}
// @lc code=end

